### PR TITLE
Use tempest-testing in misk-aws-dynamodb-testing

### PR DESCRIPTION
**Pull request by zhxnlai**
As documented here https://cashapp.github.io/tempest/guide/testing/

This PR
* does not modify public APIs in `misk-aws-dynamodb-testing` and `misk-aws2-dynamodb-testing`.
* removes`wisp-aws2-dynamodb-testing`

### CODE REVIEW

File: misk-aws-dynamodb-testing/src/main/kotlin/misk/aws/dynamodb/testing/DockerDynamoDbModule.kt [hunk 1]
```diff
@@ -30,13 +28,36 @@ class DockerDynamoDbModule(
...
 
   override fun configure() {
-    install(LocalDynamoDbModule(tables))
-    install(ServiceModule<CreateTablesService>())
-    bind<LocalDynamoDb>().toInstance(DockerDynamoDb.localDynamoDb)
+    for (table in tables) {
+      multibind<DynamoDbTable>().toInstance(table)
+    }
     multibind<HealthCheck>().to<DynamoDbHealthCheck>()
+    install(ServiceModule<TestDynamoDb>())
```
Comment (zhxnlai) [inline misk-aws-dynamodb-testing/src/main/kotlin/misk/aws/dynamodb/testing/DockerDynamoDbModule.kt:35]: this creates and cleans up tests tables

**Review by chris-ryan-square — CHANGES_REQUESTED** [review]
Needs updates to wisp aws2 dynamodb testing as well.

File: misk-aws-dynamodb-testing/src/main/kotlin/misk/aws/dynamodb/testing/DockerDynamoDbModule.kt (hunk 1 above)
Comment (chris-ryan-square) [reply misk-aws-dynamodb-testing/src/main/kotlin/misk/aws/dynamodb/testing/DockerDynamoDbModule.kt:35]: 💯

**Review by chris-ryan-square — APPROVED** [review]

**Comment by swankjesse** [comment]
I’m seeing breaks in downstream projects like this:
```
    1) [Guice/ErrorInCustomProvider]: IllegalArgumentException: `dynamoDb` is only usable while the service is running
      at InProcessDynamoDbModule.providesAmazonDynamoDB(InProcessDynamoDbModule.kt:53)
          \_ installed by: MiskTestExtension$beforeEach$module$1 -> CashFaceTestingModule -> InProcessDynamoDbModule
```

**Comment by swankjesse** [comment]
Some of the fixes to address that fallout:
 https://github.com/cashapp/backfila/pull/190
 https://github.com/squareup/cash-plasma/pull/2256

**Pull request merged**